### PR TITLE
Forge: Add Connection Profile Test

### PR DESCRIPTION
**Pull request by jrichter1**
Tests connection profile generation using sakila-h2 database.
Run with -Dtest.db={folder sakila db is located in} property to set the database up

### CODE REVIEW

**Comment by psrna** [comment]
Verification job has failed. Please fix the problem.

**Comment by jrichter1** [comment]
Java 8 API code found its way into the commit, I'll just use a Java 7 friendly code instead.

File: tests/org.jboss.tools.forge.ui.bot.test/src/org/jboss/tools/forge2/ui/bot/wizard/test/ConnectionProfileWizardTest.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+	private static final String CONNECTION_NAME = "sakila";
+	private static final String SAKILA_URL = "jdbc:h2:tcp://localhost/sakila";
+	private static final String SAKILA_USERNAME = "sa";
+	private static final String SAKILA_H2_DRIVER = "h2-1.3.161.jar";
+	private String dbFolder = System.getProperty("database.path");
+
+	@Before
+	public void prepare() {
+		DatabaseUtils.runSakilaDB(dbFolder);
```
Comment (psrna) [inline tests/org.jboss.tools.forge.ui.bot.test/src/org/jboss/tools/forge2/ui/bot/wizard/test/ConnectionProfileWizardTest.java]: Assert that dbFolder is not null.

File: tests/org.jboss.tools.forge.ui.bot.test/pom.xml [hunk 2]
```diff
@@ -44,5 +47,26 @@
...
+						<groupId>org.eclipse.tycho</groupId>
+						<artifactId>tycho-surefire-plugin</artifactId>
+					</plugin>
+				</plugins>
+			</build>
+			<properties>
+				<test.database>${test.db}</test.database>
+			</properties>
+		</profile>
+	</profiles>
```
Comment (psrna) [inline tests/org.jboss.tools.forge.ui.bot.test/pom.xml]: Please unify the property names. Otherwise it is a mess.

**Comment by psrna** [comment]
At the end, please merge multiple commits into single one.

**Comment by jrichter1** [comment]
Done, the properties are unified, and the test should be run with -Dtest.database=...

File: tests/org.jboss.tools.forge.ui.bot.test/src/org/jboss/tools/forge2/ui/bot/wizard/test/ConnectionProfileWizardTest.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+		String path = null;
+		try {
+			path = new File(dbFolder).getCanonicalPath();
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+		assertNotNull("Database path is null", path);
+
+		WizardDialog dialog = getWizardDialog("Connection: Create Profile", "(Connection: Create Profile).*");
+		new LabeledText("Connection Name:").setText(CONNECTION_NAME);
```
Comment (vpakan) [inline tests/org.jboss.tools.forge.ui.bot.test/src/org/jboss/tools/forge2/ui/bot/wizard/test/ConnectionProfileWizardTest.java]: Implement this as RedDeer dialog within o.j.t.forge.reddeer plugin please

**Comment by vpakan** [comment]
Please squash to one commit and rebase to master if necessary.

**Pull request merged**